### PR TITLE
[docs]Creating a PR for the docs as suggested

### PR DESCRIPTION
**Pull request by mgundala**
@mlvandijk raised a PR for the docs can you have a quick review on these.

### CODE REVIEW

File: content/bdd-introduction.md [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+
+   **"Behavior-driven development is about implementing an application by describing its behavior from the perspective of its stakeholders�** -- Dan North
+   
+    The main intention of a BDD framework is to avoid communication gaps between team members, foster better understanding of the customer and promote continuous communication with real world examples.
+    
+    Let me try to explain with an example, A client wanted to increase his business by developing an important feature to his software.The Client gathered all his stake holders to the meeting where they have discussions about the feature.Mostly the outcome of the discussion will be like [a blind man expalining an elephant](https://en.wikipedia.org/wiki/Blind_men_and_an_elephant). It's inevitable, however the Client reaches to one development team and explains what he really wanted. Sometimes these discussions take shape of user Stroies(in Agile world), these is the time where the developer will handover his hat to Client to describe about feature.After a succesfull description developer has his hat back on implementation.
```
Comment (mlvandijk) [inline content/bdd-introduction.md]: Remove the "Let me try to explain with an example" paragraph. Imho this paragraph does not add value in terms of explanation; the idea has already been clearly explained above.

File: content/bdd-introduction.md [hunk 2]
```diff
@@ -0,0 +1,12 @@
+## Behavior-Driven Development(BDD):
+
+   Behavior-Driven development (or BDD) is a software development technique that encourages collaboration between developers, QA and non-technical or business participants in a software project.
+
+   BDD focuses on obtaining a clear understanding of desired software behavior through discussion with stakeholders. It extends [TDD](docs/tdd-introduction) by writing test cases in a natural language that non-programmers can read. Behavior-driven developers use their native language in combination with the ubiquitous language of Domain-Driven design to describe the purpose and benefit of their code. This allow the developers to focus on why the code should be created, rather than the technical details, and minimizes translation between the technical language in which the code is written and the domain language spoken by the business, users, stakeholders, project management, etc.
```
Comment (mlvandijk) [inline content/bdd-introduction.md]: "This allow the developers" should be "This allow**s** the developers"

File: content/tdd-introduction.md [hunk 3]
```diff
@@ -0,0 +1,17 @@
+## Test Driven Development
+
+    Test-driven development (TDD) is a software development process that relies on the repetition of a very short development cycle. Requirements are turned into very specific test cases, then the software is improved to pass the new tests. This is opposed to software development that allows software to be added that isn't proven to meet requirements. Begin by writing a very small test for code that does not yet exist. Run the test and, naturally, it fails. Now write just enough code to make that test pass. Once the test passes, observe the resulting design and re-factor to remove any duplication you see. It is natural at this point to judge the design as too simple to handle all of the responsibilities this code will have.
```
Comment (mlvandijk) [inline content/tdd-introduction.md]: "re-factor to remove any duplication you see" replace with "re-factor as needed". As someone already mentioned there are many more refactorings than just DRY.

**Review by mlvandijk — CHANGES_REQUESTED** [review]
@mgundala Thanks for creating this PR! Overall a great addition.

File: content/bdd-introduction.md [hunk 4]
```diff
@@ -0,0 +1,12 @@
+## Behavior-Driven Development(BDD):
```
Comment (haroon-sheikh) [inline content/bdd-introduction.md]: can you add a space before the `(BDD)`

File: content/bdd-introduction.md [hunk 5]
```diff
@@ -0,0 +1,12 @@
...
+
+   Behavior-Driven development (or BDD) is a software development technique that encourages collaboration between developers, QA and non-technical or business participants in a software project.
+
+   BDD focuses on obtaining a clear understanding of desired software behavior through discussion with stakeholders. It extends [TDD](/tdd-introduction/) by writing test cases in a natural language that non-programmers can read. Behavior-driven developers use their native language in combination with the ubiquitous language of Domain-Driven design to describe the purpose and benefit of their code. This allows the developers to focus on why the code should be created, rather than the technical details, and minimizes translation between the technical language in which the code is written and the domain language spoken by the business, users, stakeholders, project management, etc.
```
Comment (haroon-sheikh) [inline content/bdd-introduction.md]: Typo should be `behaviour`

Comment (haroon-sheikh) [reply content/bdd-introduction.md]: Do we want `minimises` or `minimizes`?

File: content/tdd-introduction.md [hunk 6]
```diff
@@ -0,0 +1,17 @@
+## Test Driven Development
+
+    Test-driven development (TDD) is a software development process that relies on the repetition of a very short development cycle. Requirements are turned into very specific test cases, then the software is improved to pass the new tests. This is opposed to software development that allows software to be added that isn't proven to meet requirements. Begin by writing a very small test for code that does not yet exist. Run the test and, naturally, it fails. Now write just enough code to make that test pass. Once the test passes, observe the resulting design and re-factor as needed. It is natural at this point to judge the design as too simple to handle all of the responsibilities this code will have.
+
+  As the code base gradually increases in size, more and more attention is consumed by the re-factoring step. The design is constantly evolving and under constant review, though it is not predetermined. This process is known as emergent design, and is one of the most significant by-products of Test Driven Development. 
+
+To get better understanding here we go life cycle of TDD
```
Comment (haroon-sheikh) [inline content/tdd-introduction.md]: can you reword this? maybe just `TDD cycle:`

**Review by haroon-sheikh — CHANGES_REQUESTED** [review]

File: content/tdd-introduction.md [hunk 7]
```diff
@@ -0,0 +1,13 @@
+## Test Driven Development
```
Comment (olleolleolle) [inline content/tdd-introduction.md]: Hyphenation needed: "Test Driven" => "Test-Driven".

File: content/tdd-introduction.md [hunk 8]
```diff
@@ -0,0 +1,13 @@
...
+
+    Test-driven development (TDD) is a software development process that relies on the repetition of a very short development cycle. Requirements are turned into very specific test cases, then the software is improved to pass the new tests. This is opposed to software development that allows software to be added that isn't proven to meet requirements. Begin by writing a very small test for code that does not yet exist. Run the test and, naturally, it fails. Now write just enough code to make that test pass. Once the test passes, observe the resulting design and re-factor as needed. It is natural at this point to judge the design as too simple to handle all of the responsibilities this code will have.
+
+  As the code base gradually increases in size, more and more attention is consumed by the re-factoring step. The design is constantly evolving and under constant review, though it is not predetermined. This process is known as emergent design, and is one of the most significant by-products of Test Driven Development.
```
Comment (olleolleolle) [inline content/tdd-introduction.md]: Hyphenation needed: "Test Driven" => "Test-Driven".

**Comment by olleolleolle** [comment]
@mlvandijk I added tiny hyphenation notes as comments here.

**Comment by mlvandijk** [comment]
@mgundala Edited the typos, hope you don't mind. I'd still like to propose removing the paragraph "Let me try to explain with an example" as I don't think this adds value to the already clear info.
Otherwise, I'd be happy to merge (if @olleolleolle and @haroon-sheikh agree?)

**Comment by aslakhellesoy** [comment]
Hi @mgundala,

Thanks for your making your first contribution to Cucumber, and welcome to the Cucumber committers team! You can now push directly to this repo and all other repos under the cucumber organization! 🍾

In return for this generous offer we hope you will:

* ✅ Continue to use branches and pull requests. When someone on the core team approves a pull request (yours or someone else's), you're welcome to merge it yourself.
* 💚 Commit to setting a good example by following and upholding our [code of conduct](https://cucumber.io/conduct) in your interactions with other collaborators and users.
* 💬 Join the [community Slack channel](http://cucumber.io/support#slack) to meet the rest of the team and make yourself at home.
* ℹ️ Don't feel obliged to help, just do what you can if you have the time and the energy.
* 🙋 Ask if you need anything. We're looking for feedback about how to make the project more welcoming, so please tell us!

On behalf of the Cucumber core team,
Aslak Hellesøy
Creator of Cucumber

**Pull request merged**